vars/piperPipelineStageInit.groovy

```diff
@@ -183,23 +183,26 @@ void call(Map parameters = [:]) {
                 echo "[${STEP_NAME}] GitHub labels could not be retrieved from Pull Request, please make sure that credentials are maintained on multi-branch job."
             }
 
-
             setPullRequestStageStepActivation(script, config, prActions)
         }
 
         if (env.BRANCH_NAME == config.productiveBranch) {
             if (parameters.script.commonPipelineEnvironment.configuration.runStep?.get('Init')?.slackSendNotification) {
                 slackSendNotification script: script, message: "STARTED: Job <${env.BUILD_URL}|${URLDecoder.decode(env.JOB_NAME, java.nio.charset.StandardCharsets.UTF_8.name())} ${env.BUILD_DISPLAY_NAME}>", color: 'WARNING'
             }
-            if (config.inferBuildTool && env.ON_K8S) {
-                // We set dockerImage: "" for the K8S case to avoid the execution of artifactPrepareVersion in a K8S Pod.
+            Map prepareVersionParams = [script: script]
+            if (config.inferBuildTool) {
+                prepareVersionParams.buildTool = buildTool
```

Comment:
Why do we have to pass the build tool at all? If inferBuildTool it should be set by setupCommonPipelineEnvironment and thus also available in the artifactPrepareVersion  step, right?

Reply:
Yes, if the metadata defines the CPE resource as source for this parameter... lemme check.

Reply:
Nope, it currently [does not](https://github.com/SAP/jenkins-library/blob/master/resources/metadata/versioning.yaml#L85). [This](https://github.com/SAP/jenkins-library/blob/master/resources/metadata/whitesource.yaml#L43) is how it should look to make it work, I believe.

Reply:
yes